aries_cloudagent/wallet/sd_jwt.py

```diff
@@ -0,0 +1,306 @@
+        kid,
+        disclosures,
+    ):
+        """Initialize an SDJWTVerifyResult instance."""
+        super().__init__(
+            headers,
+            payload,
+            valid,
+            kid,
+        )
+        self.disclosures = disclosures
+
+
+class SDJWTVerifyResultSchema(JWTVerifyResultSchema):
+    """SDJWTVerifyResult schema."""
+
+    class Meta:
+        """SDJWTVerifyResultSchema metadata."""
+
+        model_class = SDJWTVerifyResult
+
+    disclosures = fields.List(
+        fields.List(StrOrDictField()),
+        metadata={
+            "description": "Disclosure arrays associated with the SD-JWT",
+            "example": [
+                ["fx1iT_mETjGiC-JzRARnVg", "name", "Alice"],
+                [
+                    "n4-t3mlh8jSS6yMIT7QHnA",
+                    "street_address",
+                    {"_sd": ["kLZrLK7enwfqeOzJ9-Ss88YS3mhjOAEk9lr_ix2Heng"]},
+                ],
+            ],
+        },
+    )
+
+
+class SDJWTVerifierACAPy(SDJWTVerifier):
+    """SDJWTVerifier class for ACA-Py implementation."""
+
+    def __init__(
+        self,
+        profile: Profile,
+        sd_jwt_presentation: str,
+        serialization_format: str = "compact",
+    ):
+        """Initialize an SDJWTVerifierACAPy instance."""
+        self.profile = profile
+        self.sd_jwt_presentation = sd_jwt_presentation
+        self._serialization_format = serialization_format
+
+    async def _verify_sd_jwt(self) -> SDJWTVerifyResult:
+        verified = await jwt_verify(
+            self.profile,
+            self.serialized_sd_jwt,
+        )
+        return SDJWTVerifyResult(
+            headers=verified.headers,
+            payload=verified.payload,
+            valid=verified.valid,
+            kid=verified.kid,
+            disclosures=self._disclosures_list,
+        )
+
+    def _parse_sd_jwt(self, sd_jwt):
+        if self._serialization_format == "compact":
+            (
+                self._unverified_input_sd_jwt,
+                *self._input_disclosures,
+                self._unverified_input_key_binding_jwt,
+            ) = self._split(sd_jwt)
+        else:
+            # if the SD-JWT is in JSON format, parse the json and extract the disclosures.
+            self._unverified_input_sd_jwt = sd_jwt
+            self._unverified_input_sd_jwt_parsed = json.loads(sd_jwt)
+            self._input_disclosures = self._unverified_input_sd_jwt_parsed[
+                self.JWS_KEY_DISCLOSURES
+            ]
+            self._unverified_input_key_binding_jwt = (
+                self._unverified_input_sd_jwt_parsed.get(self.JWS_KEY_KB_JWT, "")
+            )
+
+        return self._unverified_input_sd_jwt
+
+    def _create_disclosures_list(self) -> List:
+        disclosures_list = []
+        for disclosure in self._input_disclosures:
+            disclosures_list.append(
+                json.loads(self._base64url_decode(disclosure).decode("utf-8"))
+            )
+
+        return disclosures_list
```

Comment:
More a point of curiosity than anything else, what did we have to change about these methods?